pandas/core/tools/datetimes.py

```diff
@@ -495,7 +504,7 @@ def _array_strptime_with_fallback(
             # Indicates to the caller to fallback to objects_to_datetime64ns
             return None
     else:
-        if "%Z" in fmt or "%z" in fmt:
+        if any(tz is not None for tz in timezones):
```

Comment:
is the difference here for if all the strings are tznaive but we saw a tzaware datetime object?

Reply:
that's right